File: jlpt/native_ios/Sources/Data/Mappers/AuthMappers.swift
import Foundation

extension AuthResponseDTO {
    func toDomain() -> AuthSession {
        AuthSession(
            accessToken: accessToken,
            refreshToken: refreshToken,
            userId: userId,
            email: email,
            displayName: displayName,
            role: UserRole(rawValue: role) ?? .user
        )
    }
}

File: jlpt/native_android/app/src/main/java/com/jlpt/android/data/mapper/AuthMappers.kt
package com.jlpt.android.data.mapper

import com.jlpt.android.data.remote.dto.AuthResponseDto
import com.jlpt.android.domain.model.AuthSession
import com.jlpt.android.domain.model.UserRole

fun AuthResponseDto.toDomain() = AuthSession(
    accessToken = accessToken,
    refreshToken = refreshToken,
    userId = userId,
    email = email,
    displayName = displayName,
    role = runCatching { UserRole.valueOf(role) }.getOrDefault(UserRole.USER)
)
